src/arrays/programs_involving_multidimensional_data/hard/q7/MainTest.java

```diff
@@ -0,0 +1,102 @@
+package arrays.programs_involving_multidimensional_data.hard.q7;
+
+import global.BaseRandomTest;
+import global.MethodTest;
+import global.tools.CustomAssertions;
+import global.variables.Clause;
+import global.variables.clauses.*;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class MainTest extends BaseRandomTest {
+    // Java
+
+    public static final int n = 5;
+    public static final int up = 100;
```

Comment:
Can you rename these to upperBound and lowerBound please? I think the names are too unclear currently